lib/bolt_server/transport_app.rb

```diff
@@ -675,6 +679,51 @@ def make_winrm_target(target_hash)
       end
     end
 
+    # Returns the base64 encoded tar archive of plugin code that is needed to calculate
+    # custom facts
+    #
+    # @param versioned_project [String] the versioned_project to build the plugin tarball from
+    post '/project_facts_plugin_tarball' do
+      raise BoltServer::RequestError, "'versioned_project' is a required argument" if params['versioned_project'].nil?
+      content_type :json
+      in_bolt_project(params['versioned_project']) do |context|
+        # Inspired by Bolt::Applicator.build_plugin_tarball
+        sio = StringIO.new
+
+        begin
+          output = Minitar::Output.new(Zlib::GzipWriter.new(sio))
+
+          # TODO: Should this be full_modulepath
+          plugin_modulepath = context[:pal].user_modulepath
```

Comment:
See above TODO.

Reply:
Also, there is some logging in `build_plugin_tarball` -- see https://github.com/puppetlabs/bolt/blob/main/lib/bolt/applicator.rb#L315. Do we want to include that here?

Reply:
I cant imagine we would need anything from the "builtin content dir". I though at one point in bolt we explicitly did not sync anything from our packaged builtin content (the core types and providers) but maybe i am not remembering that correctly. I know we allow users to configure what they want to sync https://github.com/puppetlabs/bolt/commit/3e0fe7f39533efc99da0102d6806872daabc901b but that is out of scope here. 

I'm a bit torn, i think maybe just foll modulepath would be better for now given we dont actually ship the core types/providers that bolt does?

Reply:
as far as logging, I think logging the time it takes to pack and serialize the code would help us understand how important caching may be.

Reply:
Ok, based on discussion in Slack we'll continue using the user modulepath.

I've updated the PR with logging, had to verify that the message appears at the expected level. Not sure if that should still be trace like in build_plugin_tarball or if we want to move it up more, maybe to INFO. I guess TRACE seems fine.